provider/qemu_img_utils.py

```diff
@@ -113,7 +113,10 @@ def strace(image, trace_events=None, output_file=None):
     image_cmd = image.image_cmd
     strace_prefix = ["strace"]
     if trace_events:
-        strace_prefix.extend(("-e", ",".join(trace_events)))
+        if option:
+            strace_prefix.extend(("-e", ",".join(trace_events), ",".join(option)))
+        else:
+            strace_prefix.extend(("-e", ",".join(trace_events)))
     if output_file:
         strace_prefix.extend(("-o", output_file))
     strace_prefix = " ".join(strace_prefix)
```

Comment:
There are multiple problems regaring to your change to `strace`
1. naming issue
what's the meaning of `option`, I believe it is better to be more specific like `trace_child`.
2. why you add the check to add `-f` in the `if` block of `trace_events`?
3. where is the docstring for your new parameter?

better be like:
```
@contextlib.contextmanager
def strace(image, trace_events=None, output_file=None, trace_child=False):
    """
    Add strace to trace image related operations.
    :param image: image object
    :param trace_events: events list to trace
    :param output_file: if presented, redirect the output to file
    :param trace_child: True to enable tracing child processes with -f
    """
    image_cmd = image.image_cmd
    strace_prefix = ["strace"]
    if trace_events:
        strace_prefix.extend(("-e", ",".join(trace_events)))
    if output_file:
        strace_prefix.extend(("-o", output_file))
    if trace_child:
        strace_prefix.append("-f")
    strace_prefix = " ".join(strace_prefix)
    image.image_cmd = strace_prefix + " " + image_cmd
    try:
        yield
    finally:
        image.image_cmd = image_cmd
```

Reply:
Updated. Thanks.